src/qt/listctrl.cpp

```diff
@@ -141,96 +144,868 @@ class wxQtTreeWidget : public wxQtEventSignalHandler< QTreeWidget, wxListCtrl >
+            roles.push_back(Qt::BackgroundRole);
+        }
+
+        if ( info.GetBackgroundColour().IsOk() )
+        {
+            columnItem.m_backgroundColour = info.GetBackgroundColour().GetQColor();
+            roles.push_back(Qt::ForegroundRole);
+        }
+
+        dataChanged(modelIndex, modelIndex, roles);
+
+        return true;
+    }
+
+    void SetView(QTreeView *view)
+    {
+        m_view = view;
+    }
+
+    wxColour GetItemTextColour(long item)
+    {
+        wxCHECK_MSG(item >= 0 && static_cast<size_t>(item) < m_rows.size(),
+            wxNullColour, "Invalid row");
+
+        wxCHECK_MSG(!m_rows[item].m_columns.empty(),
+            wxNullColour, "No columns in model");
+
+        return wxColour(m_rows[item][0].m_textColour);
+    }
+
+    wxColour GetItemBackgroundColour(long item)
+    {
+        wxCHECK_MSG(item >= 0 && static_cast<size_t>(item) < m_rows.size(),
+            wxNullColour, "Invalid row");
+        wxCHECK_MSG(!m_headers.empty(),
+            wxNullColour, "No columns in model");
+
+        return wxColour(m_rows[item][0].m_backgroundColour);
+    }
+
+    wxFont GetItemFont(long item)
+    {
+        wxCHECK_MSG(item >= 0 && static_cast<size_t>(item) < m_rows.size(),
+            wxNullFont, "Invalid row");
+
+        wxCHECK_MSG(!m_headers.empty(),
+            wxNullFont, "No columns in model");
+
+        return wxFont(m_rows[item][0].m_font);
+    }
+
+    long FindItem(long start, const QString& str, bool partial) const
+    {
+        if ( start < 0 )
+            start = 0;
+
+        const QString strUpper = str.toUpper();
+
+        const long numberOfRows = m_rows.size();
+        const long numberOfColumns = m_headers.size();
+
+        if ( partial )
+        {
+            for ( long i = start; i < numberOfRows; ++i )
+            {
+                for ( long j = 0; j < numberOfColumns; ++j )
+                {
+                    const QString current = m_rows[i][j].m_label.toUpper();
+                    if ( current.contains(strUpper) )
+                        return i;
+                }
+            }
+
+            return -1;
+        }
+
+        for ( long i = start; i < numberOfRows; ++i )
+        {
+            for ( long j = 0; j < numberOfColumns; ++j )
+            {
+                if ( m_rows[i][j].m_label.toUpper() == strUpper )
+                    return i;
+            }
+        }
+
+        return -1;
+    }
+
+    long FindItem(long start, wxUIntPtr data) const
+    {
+        if ( start < 0 )
+            start = 0;
+
+        const long count = m_rows.size();
+
+        for ( long i = start; i < count; ++i )
+        {
+            if ( m_rows[i].m_data == reinterpret_cast<void*>(data) )
+                return i;
+        }
+
+        return -1;
+    }
+
+
+    long InsertItem(wxListItem info)
+    {
+        const int column = info.GetColumn();
+
+        if ( static_cast<size_t>(column) >= m_headers.size() )
```

Comment:
We should check that `column > 0` before casting it to `size_t` as it's going to be disastrous if it's ever -1 here.